frontend/lib/ui/Table/Table.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from "react";
+import { useNavigate } from "react-router-dom";
+
+import cls from "./Table.module.css";
+
+export interface TableProps {
+  id?: string;
+  children?: React.ReactNode;
+}
+
+export function Table({ id, children }: TableProps) {
+  return (
+    <table id={id} className={cls["table"]}>
+      {children}
+    </table>
+  );
+}
+
+Table.Header = Header;
+Table.Column = Column;
+Table.Body = Body;
+Table.Row = Row;
+Table.LinkRow = LinkRow;
+Table.Cell = Cell;
+
+function Header({ children }: { children: React.ReactNode[] }) {
+  return (
+    <thead>
+      <tr>{children}</tr>
+    </thead>
+  );
+}
+
+function Column({ children, number, width }: { children: React.ReactNode; number?: boolean; width?: string }) {
+  return (
+    <th className={number ? cls["number"] : undefined} style={width ? { width } : undefined}>
+      {children}
+    </th>
+  );
+}
+
+function Body({ children }: { children: React.ReactNode[] }) {
+  return <tbody>{children}</tbody>;
+}
+
+function Row({ children }: { children: React.ReactNode[] }) {
+  return <tr>{children}</tr>;
+}
+
+function LinkRow({ children, to }: { children: React.ReactNode[]; to: string }) {
+  const navigate = useNavigate();
+
+  function handleClick() {
+    navigate(to);
+  }
+
+  return (
+    <tr className={cls["row-link"]} onClick={handleClick}>
```

Comment:
We might want to consider using a different element for this - for example:

```jsx
<a role="row" className={cls["row-link"]} onClick={handleClick}>
```

And then:

```css
.row-link {
  display: table-row;
}
```

This will still be rendered correctly as a table, but the clickable element is better recognized as such by browsers.

Reply:
I believe (and checked with mdn and validator.w3.org) that the only content allowed in a `tbody` is zero or more `tr` elements. For now I propose to keep it this way, and later find another more accessible way that will be either not using `table` at all, or a link within a cell and some css trickery to make it cover the whole row.